src/Security/Security.php

```diff
@@ -444,14 +510,29 @@ public function ping()
      */
     public function logout($redirect = true)
     {
-        $member = Member::currentUser();
-        if ($member) {
-            $member->logOut();
+        $this->extend('beforeMemberLoggedOut');
+        $member = static::getCurrentUser();
+
+        if ($member) { // If we don't have a member, there's not much to log out.
+            /** @var Authenticator $authenticator */
+            $authenticator = $this->getAuthenticator('default'); // Always use the default authenticator to log out
```

Comment:
Shouldn't this be

$authenticators = $this->getAuthenticators(Authenticator::LOGOUT);

Which is the point of exposing capability-gnostic authenticators?
+++
Yes, yes it should.